packages/events/src/router/middleware/validate/index.ts

```diff
@@ -67,10 +70,24 @@ export function validateAction(actionType: ActionType) {
     )
 
     if (errors.length > 0) {
-      throw new TRPCError({
-        code: 'BAD_REQUEST',
-        message: JSON.stringify(errors)
-      })
+      if (
+        actionType === ActionType.DECLARE &&
+        inScope({ Authorization: setBearerForToken(opts.ctx.token) }, [
+          SCOPES.RECORD_SUBMIT_INCOMPLETE
+        ])
+      ) {
+        if (!opts.input.metadata) {
+          opts.input.metadata = {}
+        }
+        opts.input.metadata.incomplete = true
+        return opts.next()
+      }
+      {
+        throw new TRPCError({
+          code: 'BAD_REQUEST',
+          message: JSON.stringify(errors)
+        })
+      }
```

Comment:
Let's lift this logic up as high as we can and make the "incomplete declaration" just be the `NOTIFY` action internally. NOTIFY should allow any form field in the declaration form to be left empty. This will help us keep the DECLARE action data model strict

Reply:
```ts
  actions: router({
    notify: publicProcedure
      .use(requiresAnyOfScopes([SCOPES.RECORD_SUBMIT_INCOMPLETE]))
      .input(NotifyActionInput)
      .use(middleware.validateAction(ActionType.NOTIFY))
```

Easiest way to make notification accept more relaxed data is most likely by 

```
      .use(middleware.validateAction(ActionType.NOTIFY))
```

making a variation of the `validateAction` middleware that treats all form fields as optional